xds/internal/xdsclient/xds.go

```diff
@@ -323,8 +323,35 @@ func processNetworkFilters(filters []*v3listenerpb.Filter) ([]HTTPFilter, error)
 			}
 			if !seenHCM {
 				// TODO: Implement terminal filter logic, as per A36.
-				httpFilters = filters
+				filterChain.HTTPFilters = filters
 				seenHCM = true
+				switch hcm.RouteSpecifier.(type) {
+				case *v3httppb.HttpConnectionManager_Rds:
+					if hcm.GetRds().GetConfigSource().GetAds() == nil {
+						return nil, fmt.Errorf("ConfigSource is not ADS: %+v", hcm)
+					}
+					name := hcm.GetRds().GetRouteConfigName()
+					if name == "" {
+						return nil, fmt.Errorf("empty route_config_name: %+v", hcm)
+					}
+					filterChain.RouteConfigName = name
+				case *v3httppb.HttpConnectionManager_RouteConfig:
+					// "RouteConfiguration validation logic inherits all
+					// previous validations made for client-side usage as RDS
+					// does not distinguish between client-side and
+					// server-side." - A36
+					// Can specify v3 here, as will never get to this function
+					// if v2.
+					routeU, err := generateRDSUpdateFromRouteConfiguration(hcm.GetRouteConfig(), nil, false)
+					if err != nil {
+						return nil, fmt.Errorf("failed to parse inline RDS resp: %v", err)
+					}
+					filterChain.InlineRouteConfig = &routeU // Now that all this is here, scale up listener inputs in test to persist route update
+				case nil:
+					return nil, fmt.Errorf("no RouteSpecifier: +%v", hcm)
```

Comment:
https://github.com/grpc/proposal/pull/237 states the following:
```
If an XdsServer implementation uses RouteConfiguration or supports any HTTP
filters other than the hard-coded Router, then
`HttpConnectionManager.route_config` and `HttpConnectionManager.rds` must be
supported and RouteConfigurations must be validated. RouteConfiguration
```

So, is this correct to unconditionally expect route configuration to be present here?

Reply:
I was wondering about whether this was incorrect as well. I think your hunch is right, as I thought about this too. Switched to a no op.